tests/ecmascript/test-bi-math-tonumber.js

```diff
@@ -0,0 +1,62 @@
+/*
+ *  Math object ToNumber side effects (ES5 15.8.2)
+ *
+ *  Ecmascript 5.1 requires that all Math object methods coerce all arguments
+ *  with ToNumber from left to right.  This ensures that all side effects
+ *  caused by ToNumber coercion have a chance to be evaluated.
+ */
+
+/*===
+ABS
+ape
+CEIL
+goku
+FLOOR
+goku
+MIN/MAX
+goku
+ape
+cow
+pig
+pig
+cow
+ape
+goku
+POW
+*munch*
+*munch*
+ROUND
+goku
+SQRT
+pig
+ape
+===*/
+
+var a = { valueOf: function() { print("pig"); return 812; } };
+var b = { valueOf: function() { print("cow"); return 1208; } };
+var c = { valueOf: function() { print("ape"); return -128; } };
+var d = { valueOf: function() { print("goku"); return 9001.5; } };
+var e = { valueOf: function() { print("*munch*"); return 8; } };
+
+print("ABS");
+Math.abs(c);
+
+print("CEIL");
+Math.ceil(d);
+
+print("FLOOR");
+Math.floor(d);
+
+print("MIN/MAX");
+Math.min(d, c, b, a);
+Math.max(a, b, c, d);
+
+print("POW");
+Math.pow(e, e);
+
```

Comment:
Not sure if this pull is ready to merge, but just in case: it'd be better to use different arguments for Math.pow() so that the order was confirmed by the test.